src/coredump/coredumpctl.c

```diff
@@ -501,6 +551,10 @@ static int print_list(FILE* file, sd_journal *j, Table *t) {
                 RETRIEVE(d, l, "COREDUMP", coredump);
         }
 
+        free_and_replace(filename, (char *) { path_join(j->prefix, filename) });
+        if (!filename)
+                return log_oom();
```

Comment:
So we should probably use `chase_symlinks()` here and elsewhere actually with the `CHASE_PREFIX_ROOT` flag instead. So something like the following:

```
filename = mfree(filename);
r = chase_symlinks(filename, j->prefix, CHASE_PREFIX_ROOT, &filename, NULL);
if (r < 0)
        return log_error_errno(r, "Failed to resolve coredump: %m");
```

This takes care of resolving symlinks and all kinds of other stuff.

Reply:
I should probably reintroduce a function to encapsulate this logic. `make_filename_absolute` as proposed by @poettering